Source/Thunder/PluginServer.h

```diff
@@ -1277,16 +1279,33 @@ namespace PluginHost {
             string Substitute(const string& input) const override {
                 return (_administrator.Configuration().Substitute(input, PluginHost::Service::Configuration()));
             }
-           Core::hresult Metadata(string& info /* @out */) const override {
+            Core::hresult Metadata(string& info /* @out */) const override {
                 Metadata::Service result;
                 GetMetadata(result);
                 result.ToString(info);
                 return (Core::ERROR_NONE);
             }
+            std::vector<RPC::Object::Environment> SubstituteList(const std::vector<RPC::Object::Environment>& environments) const {
```

Comment:
Why do we want to create a "new" list. Can we not just reuse the old list ? It ssaves a lot of memory and if we call this the assumption is that there is no need to "save" the old one..

Reply:
addressed